proto/compiler.bzl

```diff
@@ -118,14 +118,15 @@ def go_proto_compile(go, compiler, protos, imports, importpath):
     args.add_all(imports, before_each = "-import")
     args.add_all(proto_paths.keys())
     args.use_param_file("-param=%s")
+    data_depsets = [target.files for target in compiler.internal.data]
```

Comment:
How does your compiler learn of the paths of these extra files? This will add them to the sandbox, but it doesn't tell the compiler or plugin where to look for them. Paths of generated files have the `bazel-out/k8-exec-ST-some-hash` bit in them which makes it impossible to hardcode them.

I'm wondering whether it would be better to add them to the runfiles of the `plugin` executable instead, which would allow the plugin to find them using their runfiles path via the Go runfiles library.

Reply:
The idea of this change is that the data targets become available during the runtime of the compiler. In my case the compiler knows where to look for a file based on the name of the proto file being processed. The file paths of the data targets are available as relative paths from the working directory when the compiler is executed.

Reply:
I see, so the data files required depend on the input files and aren't fixed for this particular plugin?

Reply:
That is correct. In my case I am doing two protoc passes. Both use the same `.proto` input files, but the second uses the output of the first as the data

Reply:
Sorry for the delay, I forgot to reply.

I'm still not sure I understand how this can work, but I guess it does :⁠-⁠) I have some reservations about merging this as the `data` attributes of executable rules almost universally end up adding *runfiles* to actions using the executable, not *input files*. I would feel much more comfortable supporting adding runfiles to the compiler.

Have you been able to try using runfiles for your purpose instead? That would also resolve the questions of how the file paths are known at execution time: runfiles paths don't contain the `bazel-out` path segments.